supervisor/core/supervisor_test.go

```diff
@@ -0,0 +1,74 @@
+package core
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test the basic Supervisor functionality.
+func TestSupervisorBase(t *testing.T) {
+	assert := assert.New(t)
+	// Create config for Supervisor.
+	cfg := new(Cfg)
+	cfg.InstancesDir = "../../test_instances"
+	cfg.TermTimeout = 100 * time.Millisecond
+
+	// Create Supervisor.
+	sv := NewSupervisor(cfg)
+
+	// Set a cleanup callback.
+	t.Cleanup(func() { sv.StopAllInstance() })
```

Comment:
This wouldn't work if `StopAllInstance` become broken.

Reply:
Yep. 
But this is the single legal way to stop all instances in the cleanup process. You cannot access the Instances map. Killing a process with something like `pkill` with using a pattern is also not the right way. I trust this function.